GitUtils.ps1

```diff
@@ -1,12 +1,42 @@
 # Inspired by Mark Embling
 # http://www.markembling.info/view/my-ideal-powershell-prompt-with-git-integration
 
+<#
+.SYNOPSIS
+    Gets the path to the current repository's .git dir.
+.DESCRIPTION
+    Gets the path to the current repository's .git dir.  Or if the repository
+    is a bare repository, the root directory of the repository.
+.EXAMPLE
+    PS C:\GitHub\posh-git\tests> Get-GitDirectory
+    Returns C:\GitHub\posh-git\.git
+.INPUTS
+    None.
+.OUTPUTS
+    System.String
+#>
 function Get-GitDirectory {
-    if ($Env:GIT_DIR) {
+    $pathInfo = Microsoft.PowerShell.Management\Get-Location
+    if (!$pathInfo -or $pathInfo.Provider.Name -ne 'FileSystem') {
+        $null
+    }
+    elseif ($Env:GIT_DIR) {
         $Env:GIT_DIR
     }
     else {
-        Get-LocalOrParentPath .git
+        if (Test-Path .git) {
+            [System.IO.Path]::Combine($pathInfo.Path, '.git')
+        }
+        else {
+            $gitDir = git rev-parse --git-dir 2>$null

```

Comment:
Interesting I don't see errors with this:
```
~\GitHub\temp\test.git [BARE:master]>
```
But in the past day or two I've seen non-terminating, error records appearing after a successful git command.  I wonder if this is why git rev-parse was abandoned in the first place?  

OTOH a non-zero exitcode should not cause PowerShell to display errors.  Normally that happens with stderr output but we are redirecting that to $null. 

Aha, you will get a NativeCommandError output above is someone has set $ErrorActionPreferencd to 'Stop'.   Can you stick in a  Write-Warning $ErrorActionPreference before the git rev-parse line and see what the value is?